src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/AsyncRestExecutor.java

```diff
@@ -66,17 +71,17 @@
     public void execute(Client client, Map<String, String> params, QueryAction queryAction, RestChannel channel) throws Exception {
         if (isBlockingAction(queryAction) && isRunningInTransportThread()) {
             if (LOG.isDebugEnabled()) {
-                LOG.debug("Async blocking query action [{}] for executor [{}] in current thread [{}]",
-                    name(executor), name(queryAction), Thread.currentThread().getName());
+                LOG.debug("[{}] Async blocking query action [{}] for executor [{}] in current thread [{}]",
```

Comment:
I think this approach of pre-pending request id to log messages does not scale + is error prone. So far I've seen this type of change in log4j configuration itself, which includes the request id in all log messages for a given appender and then you don't have to worry about it in the code.

Reply:
> I think this approach of pre-pending request id to log messages does not scale + is error prone. So far I've seen this type of change in log4j configuration itself, which includes the request id in all log messages for a given appender and then you don't have to worry about it in the code.

I'm not sure how log4j could be aware of the relation between logging entries. Because here basically we want to track all logs related to a request by an unique ID. Could you give me a sample?

Reply:
I couldn't find a good example right now, but I remember I did that previously with slf4j (using log4j implementation); in a nutshell it has some logging context in the thread context, and you just write the requestId into that as soon as you get the request, and the logger knows how to pull values from that context, you don't need to specify the value yourself in each log call. This will be a bigger change, so let's create an issue to track it and we can decide how to fix it later. Thanks!

Reply:
Sure. Thanks!

Reply:
Created issue for tracking: https://github.com/opendistro-for-elasticsearch/sql/issues/34